packages/template-retail-react-app/app/commerce-api/contexts.js

```diff
@@ -13,6 +19,26 @@ export const CommerceAPIContext = createContext()
 export const CommerceAPIProvider = CommerceAPIContext.Provider
 export const useCommerceAPI = () => useContext(CommerceAPIContext)
 
+/**
+ * Extract the React SDK providers used across the app
+ * This combines QueryClient, Config, CommerceApi, and Auth contexts into one reusable component
+ */
+export const CommerceSDKReactProvider = ({api, children}) => {
+    const queryClient = new QueryClient()
+    
+    return (
+        <QueryClientProvider client={queryClient}>
+            <ConfigContext.Provider value={api._authConfig}>
```

Comment:
One idea to integrate `willSendRequest` with the implementation in commerce-sdk-react is to implement a willSendRequest that works for both (commerce-api and commerce-sdk-react) in `commerce-api` and include it in the ConfigContext provider.
```
    const configContextValue = {
        ...api._authConfig,
        willSendRequest: api.willSendRequest
    }
<ConfigContext.Provider value={configContextValue}>

```